Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasFunction_Source.jinja

```diff
@@ -0,0 +1,178 @@
+{% endif -%}
+{% set branchwithvalue = function.attrib['BranchWithValue'] %}
+{%- if branchwithvalue is defined and branchwithvalue.lower() == "true" %}
+                branchResultInfo.m_returnResultInBranches = true;
+{% endif -%}
+{{SetBranchExecution(function)}}
+{%- endif -%}
+
+
+            {% set functionName = macros.CleanName(function.attrib['Name']) %}
+            {% set sanitizedFunctionName = macros.CleanName(function.attrib['Name']).replace('::', '_') %}
+            {
+             behaviorContext->Method("{{sanitizedNamespaceName}}_{{sanitizedFunctionName}}", &{{className}}::{{functionName}},
+{{macros.GenerateFunctionMetaData(function)}})
+                    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+{% if categoryName %}
+                    ->Attribute(AZ::Script::Attributes::Category, "{{categoryName}}")
+{% endif %}
+{% if hasbranch|booleanTrue == true %}
+                    ->Attribute(AZ::ScriptCanvasAttributes::BranchOnResult, branchResultInfo)
+{% endif %};
+            }
+        }
+        {% endfor %}
+
+
+
+        {# Reflect Data Types to Behavior Context #}
+{%      if Library.tag == "Data" %}
+{%         set dataName = Library.attrib['Name'] %}
+{%         set namespaceName = '' %}
+{%         set sanitizedNamespaceName = '' %}
+{%         if Library.attrib['Namespace'] is defined and Library.attrib['Namespace'] %}
+{%             set namespaceName = Library.attrib['Namespace'] + '::' %}
+{%             set sanitizedNamespaceName = macros.CleanName(Library.attrib['Namespace'].replace('::', '_')) + '_' %}
+{%         endif %}
+            behaviorContext->Class<ContainerTypeReflection::BehaviorClassReflection<{{namespaceName}}{{dataName}}>>("ScriptCanvas_{{sanitizedNamespaceName}}{{dataName}}")
+                ->Attribute(AZ::Script::Attributes::ExcludeFrom, AZ::Script::Attributes::ExcludeFlags::All)
+                ->Attribute(AZ::Script::Attributes::Ignore, true)
+                ->Method("ScriptCanvas_ReflectVector", [](const AZStd::vector<{{namespaceName}}{{dataName}}>&) {})
+                ->Method("ScriptCanvas_ReflectStringMap", [](const AZStd::unordered_map<AZStd::string, {{namespaceName}}{{dataName}}>&) {})
+                ->Method("ScrtptCanvas_ReflectNumberMap", [](const AZStd::unordered_map<double, {{namespaceName}}{{dataName}}>&) {});
+
+{%     endif %}
+
+
+    }
+
+
+    {# Reflect Data Types to Containers in Behavior Context #}
+{%      if Library.tag == "Data" %}
+{%         set dataName = Library.attrib['Name'] %}
+{%         set namespaceName = '' %}
+{%         if Library.attrib['Namespace'] is defined and Library.attrib['Namespace'] %}
+{%             set namespaceName = Library.attrib['Namespace'] + '::' %}
+{%         endif %}
+        ContainerTypeReflection::HashContainerReflector<{{namespaceName}}{{dataName}}>::Reflect(context);
+
+{%     endif %}
+
+
+}
+
+
+
+
+
```

Comment:
What is with all this extraspace?
I know it is autogen, but this is unneccessary
```suggestion
}

```